packages/cacti-plugin-ledger-connector-stellar/src/main/typescript/plugin-ledger-connector-stellar.ts

```diff
@@ -129,8 +132,38 @@ export class PluginLedgerConnectorStellar
     return response;
   }
 
-  public transact(): Promise<unknown> {
-    throw new InternalServerError("Method not implemented.");
+  public async runSorobanTransaction(
+    req: RunSorobanTransactionRequest,
+  ): Promise<RunSorobanTransactionResponse> {
+    const invokeArgs = {
+      ...req,
+      methodArgs: req.methodArgs ?? {},
+      networkConfig: this.networkConfig,
+      fnLogPrefix: this.className,
+      txInvocation: convertApiTransactionInvocationToStellarPlus(
+        req.transactionInvocation,
+        this.networkConfig,
+      ),
+    };
+
+    const result = req.readOnly
+      ? await invokeContract<typeof invokeArgs.methodArgs>({
+          ...invokeArgs,
+          readOnly: true,
+        })
+      : await invokeContract<typeof invokeArgs.methodArgs>({
+          ...invokeArgs,
+          readOnly: false,
+        });
```

Comment:
I think this can be splified now  when `req.readOnly` is required (or you can always do `!!req.readOnly` to force boolean from a variable even if it's undefined / null / etc...)

Reply:
Thank you @outSH !

This is actually a great suggestion to make this piece more readable!  
I'll make the adjustments right away!